<p>
A long parameter list can indicate that a new structure should be created to wrap the numerous parameters or that the function is doing too many things.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
public function addData(p1 : int, p2 : int, p3 : int, p4 : int, p5 : int): void  { // Noncompliant
...
}
</pre>

<h2>Compliant Solution</h2>
<pre>
public function addData(p1 : int, p2 : int, p3 : int, p4 : int): void  { // Compliant
...
}
</pre>
