<p>
  A listener can be attached to an object only after it has been constructed. So dispatching an event in a constructor is useless and error prone.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
public class MyClass
{
  public function MyClass()
  {
    dispatchEvent( new Event( "uselessEvent" ) ); }
  }
}
</pre>
