<p>
  Try to imagine using the standard Flex API without ASDoc. It would be a nightmare, because ASDoc is the only way to understand of the contract of the API.</br>
  Documenting an API with ASDoc increases the productivity of the developers use it.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
public class MyClass {
  public var myLabel:String;

  public function myMethod(param1:String):Boolean {...}
}
</pre>

<h2>Compliant Solution</h2>
<pre>
/**
 * my doc
 */
public class MyClass {
  /**
   * my doc
   */
  public var myLabel:String;

  /**
   * my doc
   * @param param1 my doc
   * @return my doc
   */
  public function myMethod(param1:String):Boolean {...}
}
</pre>

<h2>Exceptions</h2>
<p>Classes or class elements with an ASDoc <code>@private</code> comment are ignored by this rule.</p>
<pre>
/**
 * @private  // This class and all its elements are ignored
 */
public class MyClass {  // Compliant

  public var myLabel:String;   // Compliant
}

public class AnotherClass {  // Noncompliant; class not @private and not documented

  /**
   * @private
   */
  public var name:String;  // Compliant
}
</pre>
